src/base/bittorrent/trackerentry.cpp

```diff
@@ -52,59 +52,56 @@ QString TrackerEntry::url() const
     return QString::fromStdString(nativeEntry().url);
 }
 
-bool TrackerEntry::isWorking() const
-{
-    // lt::announce_entry::is_working() returns
-    // true when the tracker hasn't been tried yet.
-#if (LIBTORRENT_VERSION_NUM < 10200)
-    return nativeEntry().verified && nativeEntry().is_working();
-#else
-    if (!nativeEntry().verified)
-        return false;
-    const auto &endpoints = nativeEntry().endpoints;
-    return std::any_of(endpoints.begin(), endpoints.end()
-                       , [](const lt::announce_endpoint &endpoint)
-    {
-        return endpoint.is_working();
-    });
-#endif
-}
-
 int TrackerEntry::tier() const
 {
     return nativeEntry().tier;
 }
 
 TrackerEntry::Status TrackerEntry::status() const
 {
-    if (isWorking())
-        return Working;
-
 #if (LIBTORRENT_VERSION_NUM < 10200)
-    if ((nativeEntry().fails == 0) && nativeEntry().updating)
+    if (nativeEntry().fails > 0)
+        return NotWorking;
+
+    if (nativeEntry().updating)
         return Updating;
-    if (nativeEntry().fails == 0)
-        return NotContacted;
 #else
     const auto &endpoints = nativeEntry().endpoints;
-    const bool allFailed = std::all_of(endpoints.begin(), endpoints.end()
-                                       , [](const lt::announce_endpoint &endpoint)
+
+    const bool allFailed = !endpoints.empty() && std::all_of(endpoints.begin(), endpoints.end()
```

Comment:
It is possible that there is no endpoint here?

Reply:
Yes, I saw it empty when testing the code.

Reply:
IMO, in this case we should assign some status explicitly (one of the existing, e.g. NotContacted, or the new one, e.g. NotAvailable).

Reply:
When `endpoints` are empty, it will eventually fall into the `NotContacted` state. I think this is good enough than explicitly assigning a state to it.